ansible/roles/real-postfix/templates/main.cf.j2

```diff
@@ -0,0 +1,56 @@
+#biff = no
+
+# appending .domain is the MUA's job.
+append_dot_mydomain = no
+
+# Uncomment the next line to generate "delayed mail" warnings
+#delay_warning_time = 4h
+
+alias_maps = hash:/etc/aliases
+alias_database = hash:/etc/aliases
+myorigin = scripts.mit.edu
+mydestination = scripts.mit.edu, scripts, $myhostname, scripts-test.mit.edu, scripts-test, localhost
+relayhost =
+mynetworks_style = host
+mailbox_command_maps =
+    texthash:/etc/postfix/mailbox_command_maps,
+    ldap:/etc/postfix/mailbox-command-maps-ldap.cf
+mailbox_size_limit = 0
+message_size_limit = 41943040
+recipient_delimiter = +
+inet_interfaces = $myhostname{% for vip in vips %}{% if vip.mail | default(False) and (vip.codename|default(ansible_lsb.codename) == ansible_lsb.codename) %}, {{ vip.host }}{% endif %}{% endfor %}
+
+readme_directory = /usr/share/doc/postfix/README_FILES
+sample_directory = /usr/share/doc/postfix/samples
+sendmail_path = /usr/sbin/sendmail
+html_directory = no
+setgid_group = postdrop
+command_directory = /usr/sbin
+manpage_directory = /usr/share/man
+daemon_directory = /usr/libexec/postfix
+newaliases_path = /usr/bin/newaliases
+mailq_path = /usr/bin/mailq
+queue_directory = /var/spool/postfix
+mail_owner = postfix
+# virtual-alias-maps-ldap-reserved.cf redirects {abuse,webmaster,etc.}@ -> root
+# For domains on this pool, virtual-alias-maps-ldap.cf does foo@bar.com -> user+foo@localhost
+# For domains on another pool, virtual-alias-maps-relay-ldap.cf does foo@bar.com -> foo!bar.com@[18.4.86.22]
+# For users on another pool, pass-scripts.mit.edu matches *@scripts.mit.edu and virtual-alias-maps-relay-user-ldap.cf does foo@scripts.mit.edu -> foo!scripts.mit.edu@[18.4.86.22]
+# pass-scripts.mit.edu-suffix and virtual-alias-maps-relay-user-suffix-ldap.cf do the same for foo+bar@scripts.mit.edu
+# Native routing does foo@scripts.mit.edu -> foo
+# smtp_generic_map rewrites foo!scripts.mit.edu@[18.4.86.22] -> foo@scripts.mit.edu after the transport is selected
+# To override, put something in /etc/postfix/force_pool
+virtual_alias_domains = {% for vip in vips | rejectattr('type', 'defined') %}{% if (vip.codename|default(ansible_lsb.codename) == ansible_lsb.codename) %}!{{ vip.host }}, !{{ vip.host | replace('.mit.edu', '') }}, {% endif %}{% endfor %}!$myhostname, !localhost, pcre:/etc/postfix/force_pool, ldap:/etc/postfix/virtual-alias-domains-ldap.cf
```

Comment:
We have never accepted mail to "scripts-vhosts" (the name, not the IP). I do not think that should change. Especially since there is the canonicalization behavior you mentioned, it would not be good if "quentin@miriam.mit.edu" suddenly got delivered to me instead of miriam.

And yeah, if mail comes in externally to the wrong pool, it will get rejected. If it comes in locally for -f20, it should get forwarded like any other non-local mail.

Reply:
> We have never accepted mail to "scripts-vhosts" (the name, not the IP). I do not think that should change. Especially since there is the canonicalization behavior you mentioned, it would not be good if "quentin@miriam.mit.edu" suddenly got delivered to me instead of miriam.

As in, if some servers canonicalize and some don't, we'd rather that be "reject" or "accept to miriam's locker", not "accept for Quentin" or "accept for Miriam"? Makes sense.

I think I'm sorta in favor of a comment explaining some of this, but whatever.